EventListener/AttachRefreshTokenOnSuccessListener.php

```diff
@@ -107,9 +121,27 @@ public function attachRefreshToken(AuthenticationSuccessEvent $event): void
             $refreshToken = $this->refreshTokenGenerator->createForUserWithTtl($user, $this->ttl);
 
             $this->refreshTokenManager->save($refreshToken);
-            $data[$this->tokenParameterName] = $refreshToken->getRefreshToken();
+            $refreshTokenString = $refreshToken->getRefreshToken();
         }
 
-        $event->setData($data);
+        if (isset($this->cookieSettings['enabled']) && $this->cookieSettings['enabled']) {
+            $event->getResponse()->headers->setCookie(
+                new Cookie(
+                    $this->tokenParameterName,
+                    $refreshTokenString,
+                    time() + $this->ttl,
+                    $this->cookieSettings['path'],
+                    $this->cookieSettings['domain'],
+                    $this->cookieSettings['secure'],
+                    $this->cookieSettings['http_only'],
+                    false,
+                    $this->cookieSettings['same_site']
+                )
+            );
+        } else {
+            $data[$this->tokenParameterName] = $refreshTokenString;
+
+            $event->setData($data);
```

Comment:
If I understand correctly you would like to be able to use both `refresh_token` in response cookie as well as the `JsonResponse`?

In my opinion, since obtaining a `refresh_token` is enough to impersonate a user and gain indefinite access to the application, it is paramount to ensure the `refresh_token` is not stolen. By setting it in the `JsonResponse`, JavaScript executed in the browser will be able to read the `refresh_token` from the response, while JavaScript would not be able to read the value from a HttpOnly cookie.

I think, if cookies are enabled, by default it should not send the `refresh_token` in the `JsonResponse` to protect from this scenario.

I will have a look if I can do something similar to https://github.com/lexik/LexikJWTAuthenticationBundle/issues/909 to allow users to explicitly enable this behavior if it is really needed.

Reply:
@djpremier Thanks for the suggestion, [this](https://github.com/markitosgv/JWTRefreshTokenBundle/pull/274/commits/d4bd2b1f9428c593d49b05d561006f40a11a4204) should do what you asked for.

```
gesdinet_jwt_refresh_token:
    cookie:
      enabled: true
      remove_token_from_body: false
```

Reply:
Hi @Jayfrown, yes I also agree that the best is to use the cookie, but as raised in the issue, if the application is geared towards a globally-used API (via command line for example), or in the case of a mobile application, have access to the refresh_token is required.

Thank you very much for adding the functionality, the fact that it is optional is the best, because if you ever need to use it, it's simple.